lib/rules/spaced-comment.js

```diff
@@ -281,26 +328,36 @@ module.exports = {
 
             // Checks.
             if (requireSpace) {
-                if (!rule.regex.test(node.value)) {
+                if (!rule.beginRegex.test(node.value)) {
```

Comment:
actually - `hasBeginningSpace` and `hasEndSpace` don't make sense if you're using `.exec()`, since that returns an array of matches instead of a boolean